core/src/lightclient/mod.rs

```diff
@@ -16,8 +28,25 @@ pub enum LightClientError {
     OtherError(#[from] anyhow::Error),
 }
 
-pub struct StarknetClient {
+#[automock]
+#[async_trait]
+pub trait StarknetClient: Send + Sync {
+    async fn block_number(&self) -> Result<u64, LightClientError>;
+    async fn get_eth_block_from_starknet_block(
+        &self,
+        block_id: StarknetBlockId,
+        hydrated_tx: bool,
+    ) -> Result<RichBlock, LightClientError>;
+    async fn get_code(
+        &self,
+        ethereum_address: Address,
+        starknet_block_id: StarknetBlockId,
+    ) -> Result<Bytes, LightClientError>;
+}
+pub struct StarknetClientImpl {
```

Comment:
Why does a struct have suffix impl in its name:)?

Reply:
Its just to separate from the trait, it is a pattern that was implemented in Beerus to